src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSourceConfig.java

```diff
@@ -0,0 +1,206 @@
+  @Macro
+  @Nullable
+  final Long scanTimeRangeStop;
+
+  @Name(ON_ERROR)
+  @Description("How to handle error in record processing. " +
+    "Error will be thrown if failed to parse value according to provided schema.")
+  @Macro
+  final String onError;
+
+  @Name(SCHEMA)
+  @Macro
+  @Description("The schema of the table to read.")
+  final String schema;
+
+  public BigtableSourceConfig(String table, String instance, @Nullable String project,
+                              @Nullable String serviceFilePath, @Nullable String keyAlias,
+                              @Nullable String scanRowStart, @Nullable String scanRowStop,
+                              @Nullable Long scanTimeRangeStart, @Nullable Long scanTimeRangeStop,
+                              String onError, String schema) {
+    this.table = table;
+    this.instance = instance;
+    this.project = project;
+    this.serviceFilePath = serviceFilePath;
+    this.keyAlias = keyAlias;
+    this.scanRowStart = scanRowStart;
+    this.scanRowStop = scanRowStop;
+    this.scanTimeRangeStart = scanTimeRangeStart;
+    this.scanTimeRangeStop = scanTimeRangeStop;
+    this.onError = onError;
+    this.schema = schema;
+  }
+
+  public void validate() {
+    super.validate();
+    if (!containsMacro(PROJECT) && null == project) {
+      throw new InvalidConfigPropertyException("Could not detect Google Cloud project id from the environment. " +
+                                                 "Please specify a project id.", PROJECT);
+    }
+    if (!containsMacro(INSTANCE) && Strings.isNullOrEmpty(instance)) {
+      throw new InvalidConfigPropertyException("Instance ID cannot be null or empty", INSTANCE);
+    }
+    String serviceAccountFilePath = getServiceAccountFilePath();
+    if (!containsMacro(SERVICE_ACCOUNT_FILE_PATH) && null != serviceAccountFilePath) {
+      File serviceAccountFile = new File(serviceAccountFilePath);
+      if (!serviceAccountFile.exists()) {
+        throw new InvalidConfigPropertyException(String.format("File '%s' does not exist", serviceAccountFilePath),
+                                                 SERVICE_ACCOUNT_FILE_PATH);
+      }
+    }
+    if (!containsMacro(ON_ERROR) && null != onError) {
+      if (Strings.isNullOrEmpty(onError)) {
+        throw new InvalidConfigPropertyException("On Record Error cannot be null or empty", ON_ERROR);
+      }
+      if (null == ErrorHandling.fromDisplayName(onError)) {
+        throw new InvalidConfigPropertyException("Invalid record error handling strategy name", ON_ERROR);
+      }
+    }
+    if (!containsMacro(SCHEMA)) {
+      Schema parsedSchema = getSchema();
+      if (null == parsedSchema) {
+        throw new InvalidConfigPropertyException("Schema cannot be null", SCHEMA);
+      }
+      List<Schema.Field> fields = parsedSchema.getFields();
+      if (null == fields) {
+        throw new InvalidConfigPropertyException("Schema should contain fields to map", SCHEMA);
```

Comment:
This can only happen if the schema is not a Record. We generally don't check this, though it's not wrong to have it. The message should be:

Schema is of invalid type '%s'. The schema must be a record.

Reply:
There is a method in Schema class that creates a Schema with record type. Added both checks.